### PR TITLE
Feature: Added dummy damage System.

### PR DESCRIPTION
**Pull request by ruchithdhoni**
1. Added a dummy prefab.
2. The dummy gets deleted when gun starts shooting.

### CODE REVIEW

File: Disfunction/Assets/DamageSystem.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    public GameObject dummy;
+
+    private void Start()
+    {
+        spawn(health);
+    }
+
+    public void spawn(float health)
+    {
+        float x = Random.Range(-3,5);
```
Comment (AshishPhilip999) [inline Disfunction/Assets/DamageSystem.cs:20]: Make the Spawn range X and Y as Global public variables.

Comment (DisolveStudios) [inline Disfunction/Assets/DamageSystem.cs:20]: Make the X and Y range values public global variables, so we can edit in inspector.

File: Disfunction/Assets/Game Scripts/Development/SpawnSYS/MovingDummy.cs
Comment (AshishPhilip999) [inline Disfunction/Assets/Game Scripts/Development/SpawnSYS/MovingDummy.cs:1]: Remove this and all the files that are just empty or not in use

File: Disfunction/Assets/Game Scripts/Development/SpawnSYS/Dummy.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Dummy : MonoBehaviour
+{
+    public float health;
+    public DummyDamageSystem dummyDamageSystem;
+    public bool dummyMovement;
```
Comment (DisolveStudios) [inline Disfunction/Assets/Game Scripts/Development/SpawnSYS/Dummy.cs:9]: Remove this unused boolean

File: Disfunction/Assets/Game Scripts/Development/SpawnSYS/Dummy.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        transform.Translate(direction * movementSpeed * Time.deltaTime);
+        
+
+        if (transform.position.x <= minX || transform.position.x >= maxX)
+        {
+            direction = -direction;  
+        }
+
+        // Check for collision in the forward direction and change direction if needed
+        if (Physics.Raycast(transform.position, direction, 1f))
```
Comment (DisolveStudios) [inline Disfunction/Assets/Game Scripts/Development/SpawnSYS/Dummy.cs:44]: Move this if to above if and combine it in one if statement with or (||) operator

**Comment by DisolveStudios** [comment]
organise variables in scripts. Maintain this order: 
1) Vectors
2) Numerics [float -> Double -> Int]
3) Booleans
4) Other objects and ref types

Public identifiers first then private.

**Comment by DisolveStudios** [comment]
Dont' forget to rebase to main

**Review by AshishPhilip999 — APPROVED** [review]

**Pull request merged**